TGUI for smartfridges, drying racks and disk compartmentalizer

vuonojenmustaturska opened the pull request:
[Changelogs]: 
[]: 


:cl: Naksu
add: Added TGUI interfaces to various smartfridges of different kinds, drying racks and the disk compartmentalizer
/:cl:

[why]: 
The black text on white background UI is terrible, this one lets you pull 1 or n items from storage.

vuonojenmustaturska committed:
WIP

vuonojenmustaturska committed:
WIP

vuonojenmustaturska committed:
WIP

vuonojenmustaturska committed:
WIP

MrsTonedOne committed:
Note fading and persistent messages
Player Notes can now fade out over time so admins don't accidentally ban players for 4 year old notes
Deleted messages now remain in the database and are instead just hidden from view.

Two things relating to messages I've been meaning to do for ever.

vuonojenmustaturska committed:
stuff

vuonojenmustaturska committed:
stuff

vuonojenmustaturska committed:
stuff

vuonojenmustaturska committed:
stuff

pubby committed:
Small PubbyStation update

imtakingabreakdontatme committed:
Ceiling Broke

imtakingabreakdontatme committed:
Sound

imtakingabreakdontatme committed:
Slime people can eat anything

ShizCalev committed:
Fixed Omega comms power

Commit:
Revert "Automatic unbuckle by player moving"

pubby committed:
Removes the multiverse

Commit:
Revert "Revert "Rearranges the antagonist panel, cleans it up code-wise and appearance-wise (#30255)" (#30289)"

This reverts commit 34b39c6c01eb2a64a92994f6bbb9f4b7b7127f5a.

vuonojenmustaturska committed:
seems to work now

vuonojenmustaturska committed:
seems to work now

vuonojenmustaturska committed:
seems to work now

vuonojenmustaturska committed:
seems to work now

vuonojenmustaturska committed:
Rebased tgui.js

vuonojenmustaturska committed:
Merge branch 'tgui' of https://github.com/vuonojenmustaturska/tgstation into tgui

# Conflicts:
#	tgui/assets/tgui.js

vuonojenmustaturska commented:
New UI looks like this:
![image](https://user-images.githubusercontent.com/20017308/29896312-34c33c38-8de5-11e7-9fc4-c2e40389ddba.png)

Drying racks have a dry/stop drying toggle button on the top right corner

Cyberboss committed:
Merge pull request #30281 from MoreRobustThanYou/revert-30239-dont_be_a_bitch_kingofkosmos

Revert "Automatic unbuckle by player moving"

pubby committed:
Crew monitoring rework part 1: crew pinpointers (#30117)

* Pinpointer refactor + Crew pinpointers

* Icon update

* Add crew pinpointers to maps, closets, and fix

* Fix defines

* Cyberboss fixes + add to warden and CMO

* More Cyberboss fixes

tgstation-server committed:
Automatic changelog generation for PR #30117 [ci skip]

Cyberboss committed:
Merge pull request #30277

Cyberboss committed:
Merge pull request #30273

Cyberboss committed:
Merge pull request #30266

Cyberboss committed:
Merge pull request #30257

tgstation-server committed:
Automatic changelog generation for PR #30273 [ci skip]

Cyberboss committed:
Changelog compile, [ci skip]

ShizCalev commented:
Change the wording to "Dispense One" / "Dispense (X)" instead imo

vuonojenmustaturska commented:
Regrettably, the buttons have a character limit of roughly 9 letters. You can actually see the "take many" text starting to overflow from the button already :(

Cruix committed:
Replaced datacore mob reference with a mind reference

ExcessiveUseOfCobblestone commented:
tfw no one maps disk compartmentalizers in their map yet

ShizCalev commented:
`Vend One` & `Vend (X)` then?

That way it preserves the wording from their current interfaces, as well as making it match with the seed extractor & all other vending machines.

Commit:
Sanity checkin'

MrsTonedOne committed:
Make the fact there are hidden notes more obvious.
We print the first note that would be skipped then pixel shift the show all button over the skipped noted.

The rest after that are not printed

MrsTonedOne committed:
Actually, lets make it say how many are hidden.

RemieRichards commented:
@vuonojenmustaturska make "Take" the header, and just have "One" and "Many" as the buttons

Jordie0608 committed:
unruntimes fdel on savefiles

RemieRichards committed:
UI objects should PIXEL_SCALE

vuonojenmustaturska commented:
yeah that works

vuonojenmustaturska commented:
It would look better if it was neatly aligned between the buttons, but them's the breaks I guess

![image](https://user-images.githubusercontent.com/20017308/29915100-692e78e4-8e43-11e7-9257-39e8bb2d312b.png)

vuonojenmustaturska committed:
UI changes

vuonojenmustaturska committed:
UI changes

RemieRichards commented:
Try putting the buttons in the same cell? then you don't need two headers, and it should align in the middle.

vuonojenmustaturska commented:
Putting them in the same cell makes them go on two rows with no spacing between them, and blows up the vertical space taken up by each item row. It was one of my first attempts at getting this shit done. I could align the text to either the right side of the first column or the left side of the second column, but that would look possibly worse.

I guess if I studied the TGUI CSS a bit more I could figure out how to get it done, with plain HTML it would be easy.

RemieRichards commented:
You can use plain HTML tables (any plain HTML actually, so long as it matches whatever version we shoot for), they work fine and are already used in other tguis.

vuonojenmustaturska commented:
I'll give it a try after I leave work

FuryMcFlurry committed:
add costume on mob

FuryMcFlurry committed:
added icon files for costumes

FuryMcFlurry committed:
added code for costume hats

FuryMcFlurry committed:
added code for costume jumpsuits

FuryMcFlurry committed:
added code for costume suits

AnturK committed:
Fixes no drop implant

AnturK committed:
Fixes dummy unequip runtimes. (#30285)

XDTM committed:
Prevents hallucinations revealing people's identity

Cyberboss committed:
Removes a redundant check (#30275)

FuryMcFlurry committed:
0 to FALSE

XDTM committed:
Disease Outbreak event may now also spread randomly generated advanced viruses

XDTM committed:
fix

AnturK committed:
Fixes mecha. Also adjacent called from border objects.

AnturK committed:
Why i even did that

lzimann committed:
Merge pull request #30283 from pubby/multiverse

Removes the multiverse and its sword

tgstation-server committed:
Automatic changelog generation for PR #30283 [ci skip]

ShizCalev committed:
Fixed pubby emergency shuttle (#30313)

ChangelingRain committed:
Fixes shield blobs not blocking atmos (#30291)

Iamgoofball committed:
Revert "Lowers portable pump max pressure to 1013.25"

vuonojenmustaturska commented:
It is advisable to not look at the .ract.

![image](https://user-images.githubusercontent.com/20017308/29940872-435aa970-8e99-11e7-97d7-a02f4e95300c.png)

vuonojenmustaturska committed:
Table header alignment changes

vuonojenmustaturska committed:
Table header alignment changes

Cyberboss committed:
Move along

Cyberboss committed:
Merge pull request #30292 from Xhuis/revert-30289-RevertAntag

Rearranges the antagonist panel: now with 150 more hours of testing

Commit:
Removes gangs (#30056)

tgstation-server committed:
Automatic changelog generation for PR #30056 [ci skip]

tortellinitony committed:
Makes round-end credits configurable

tortellinitony committed:
TRUEFALSE and changed the logic

ShizCalev committed:
Makes APC cell_type actually pick a cell

ShizCalev committed:
Changed to subtypes

ShizCalev committed:
Duplicate var

XDTM committed:
Logs Voice of God in the feedback database

XDTM committed:
RIP VA

YPOQ committed:
Various Ninja Fixes (#30102)

* Fixes energy net

* Ninjaboost usable while unconscious, remove unused action

* remove delay from adrenline message

* remove redundant stun check

* fixes ninja action icons

* remove toggle special interaction verb

* Remoeves slotting research disks into suit

* non ninjas can now start the suit initilisation

* remove unused code, capacity > charge

* radium now restores adrealine boosts, other suit reagents removed

* oops

* adrenaline boost removes stuttering

* ninjas no longer start with flashlights

* ninja suit deinitializes properly

* fix gib ninja suit gib message

* energy nets use buckling instead of anchoring, buckle_lying = -1 is now handled properly

* suit examine reports capcity properly

* ninja code cleanup

* undo buckle_lying changes

tgstation-server committed:
Automatic changelog generation for PR #30102 [ci skip]

Cyberboss commented on code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:
```diff
@@ -142,69 +142,52 @@
-	var/amount = text2num(href_list["amount"])
+	var/listofitems = list()
+	for (var/atom/movable/O in contents)
+		if (listofitems[O.name])
```
```
for(var/I in src)
    var/atom/movable/O = I
```

Cyberboss commented on code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:
```diff
@@ -142,69 +142,52 @@
-		return
-	usr.set_machine(src)
+/obj/machinery/smartfridge/ui_data(mob/user)
+	var/list/data = list()
```
. = list() and replace all `data`s with `.`

Cyberboss commented on code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:
```diff
@@ -142,69 +142,52 @@
+					desired--
+		else
+			return TRUE
+	return TRUE
```
you only return true if your `ui_act()` did something. So replace the last 3 lines with
```
            return TRUE
    return FALSE
````

Cyberboss commented on code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:
```diff
@@ -240,20 +223,32 @@
-		user << browse("<HEAD><TITLE>[src] supplies</TITLE></HEAD><TT>[dat]</TT>", "window=smartfridge")
-	onclose(user, "smartfridge")
+/obj/machinery/smartfridge/drying_rack/ui_data(mob/user)
+	var/list/data = list()
```
. = list() etc...

Cyberboss commented on code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:
```diff
@@ -240,20 +223,32 @@
-	updateUsrDialog()
-	update_icon()
+	var/listofitems = list()
+	for (var/atom/movable/O in contents)
```
for(var/I in src) etc...

Cyberboss commented on code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:
```diff
@@ -142,69 +142,52 @@
-	updateUsrDialog()
+/obj/machinery/smartfridge/ui_act(action, params)
+	if(..())
+		return
```
Change this to
```
. = ..()
if(.)
    return
```

Cyberboss commented on code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:
```diff
@@ -240,20 +223,32 @@
+	data["isdryer"] = TRUE
+	data["drying"] = drying
+
+	return data
```
remove this line after the `.` conversion

Cyberboss commented on code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:
```diff
@@ -240,20 +223,32 @@
+	return data
+
+/obj/machinery/smartfridge/drying_rack/ui_act(action, params)
+	switch(action)
```
Change this to
```
. = ..()
if(.)
    return
```

Cyberboss commented on code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:
```diff
@@ -240,20 +223,32 @@
+		else
+			if(..())
+				return
+	return TRUE
```
These 4 lines to
```
            return TRUE
    return FALSE
```

Cyberboss commented on code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:
```diff
@@ -142,69 +142,52 @@
+			if (params["amount"])
+				desired = text2num(params["amount"])
+			else
+				desired = input("How many items?", "How many items would you like to take out?", 1) as null|num
```
This should be done in tgui itself, otherwise it's too many sanity checks to add here

vuonojenmustaturska replied:
Doing it in TGUI is insanity

Cyberboss commented on code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:
```diff
@@ -142,69 +142,52 @@
+			else
+				desired = input("How many items?", "How many items would you like to take out?", 1) as null|num
+
+			for(var/obj/item/O in contents)
```
for(var/obj/item/O in src)

Cyberboss commented on code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:
```diff
@@ -142,69 +142,52 @@
+				if(desired <= 0)
+					break
+				if(O.name == params["name"])
+					O.loc = src.loc
```
O.forceMove(drop_location())

Cyberboss commented on code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:
```diff
@@ -142,69 +142,52 @@
+	data["name"] = name
+	data["isdryer"] = FALSE
 
+	return data
```
remove this after the `.` conversion

Cyberboss committed:
Remove old PIXEL_SCALE define

Cyberboss committed:
Merge pull request #30309

Cyberboss committed:
Merge pull request #30306

Cyberboss committed:
Merge pull request #30303

Cyberboss committed:
Merge pull request #30302

Cyberboss committed:
Merge pull request #30301

Cyberboss committed:
Merge pull request #30295

Cyberboss committed:
Merge pull request #30247

tgstation-server committed:
Automatic changelog generation for PR #30247 [ci skip]

tgstation-server committed:
Automatic changelog generation for PR #30303 [ci skip]

Cyberboss committed:
Changelog compile, [ci skip]

Commit:
Adds a new brain damage line

Commit:
Update brain_damage_lines.json

Commit:
Update brain_damage_lines.json

Cyberboss committed:
Merge pull request #30304

vuonojenmustaturska committed:
Requested changes, also made the drying rack not consume power

vuonojenmustaturska committed:
Requested changes, also made the drying rack not consume power

vuonojenmustaturska commented:
![image](https://user-images.githubusercontent.com/20017308/29982578-03504fca-8f5b-11e7-8e25-b25b80328774.png)

The "take stuff out" verb is now configurable.

MrsTonedOne committed:
More admin bans for the ban god (#30319)

pubby committed:
Pinpointer runtime fixes

AnturK committed:
Fixes airless shuttles

Cyberboss committed:
Fixes modify variables interpreting dirs wrongly (#30249)

Cyberboss commented on code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:
```diff
@@ -142,69 +142,55 @@
+			else
+				desired = input("How many items?", "How many items would you like to take out?", 1) as null|num
+
+			if (!src || !istype(src) || !usr || !istype(usr) || !Adjacent(usr) || istype(usr, /mob/dead)) // Sanity checkin' in case stupid stuff happens while we wait for input()
```
if(QDELETED(src) || QDELETED(user) || !usr.Adjacent(src))

praisenarsie committed:
Station flavour names replaced with strings lists! (#30341)

* station_prefixes

* station_names.txt

* station_suffixes.txt

* greek_letters

* Rename greek_letters to greek_letters.txt

* phoenetic_alphabet.txt

* numbers_as_words.txt

* adds the strings lists

* + ""

* removes newline

* removes newline

* removed omega and delta from station_names.txt

tgstation-server committed:
Automatic changelog generation for PR #30341 [ci skip]

Cyberboss committed:
Merge pull request #30339

Cyberboss committed:
Merge pull request #30332

Cyberboss committed:
Merge pull request #30327

Cyberboss committed:
Merge pull request #30325

Cyberboss committed:
Merge pull request #30317

Cyberboss committed:
Merge pull request #30307

tgstation-server committed:
Automatic changelog generation for PR #30307 [ci skip]

tgstation-server committed:
Automatic changelog generation for PR #30325 [ci skip]

Cyberboss committed:
Changelog compile, [ci skip]

Cyberboss committed:
Add FuryMcFlurry to Credits (#30340)

AnturK committed:
Fixes BYOS (#30356)

* Fixes BYOS

* Uses defines on dock checks.

RemieRichards committed:
Mappers can now set up quantum pads to auto-link (#30324)

XDTM committed:
Fixes vaccines (#30357)

lzimann committed:
Merge pull request #30360 from AnturK/isthisreallythathardtoremembertocheckoverrideswhenaddingnewparameter

Fixes airless shuttles

lzimann committed:
Merge pull request #30345 from pubby/pinfix

Pinpointer runtime fixes

silicons committed:
Update augments_internal.dm (#30362)

as334 committed:
Adds a mass spectrometer to the detectives closet (#30367)

tgstation-server committed:
Automatic changelog generation for PR #30367 [ci skip]

XDTM committed:
Fixes storage deletion bug, again (#30355)

* Fixes storage deletion bug, again

* non-hacky fix

YPOQ committed:
fixes state laws on department channels (#30368)

YPOQ committed:
Fixes emag_act for several machines (#30371)

pubby committed:
Fix kitty ear color (#30374)

pubby committed:
Fix monastery shuttle + Pods (#30375)

tgstation-server committed:
Automatic changelog generation for PR #30375 [ci skip]

pubby committed:
Soil plot fixes (#30378)

* Make soil plots NODECONSTRUCT

* Removes the circuit from soil

vuonojenmustaturska committed:
Requested changes

vuonojenmustaturska committed:
Requested changes

vuonojenmustaturska committed:
Getting rid of github fuckery

vuonojenmustaturska committed:
Getting rid of github fuckery

pubby committed:
Bee fixes (#30386)

pubby committed:
Make puddles not destruct into metal (#30381)

* Make puddles INDESTRUCTIBLE

* Puddles are not invincible but they aren't metal either.

Commit:
Rewrites some Orion Trail grammar (#30331)

* Changes Orion Trail grammar

* in which a grammar fix PR uses improper grammar

* you saw nothing

* grammar is important

* memory is also important

* cyberboss more like cyberTOSS

tgstation-server committed:
Automatic changelog generation for PR #30331 [ci skip]

Cyberboss committed:
Makes the config dir a variable (#30188)

* Makes the config dir a variable

* Reeee

* REEEEE

Cyberboss committed:
Makes edits to the config tree autotag "Config Update" (#30314)

ChangelingRain committed:
Soft crit, except not terrible (#30262)

* Soft crit, except a little bit harder.

* tweak

* :ok_hand:

* why was this even in here

* no radios in critical

* fix that too

* keep to original logic

* not very smart, really

* tip: do nothing tgui-side that you can do code-side, because we can't do defines in tgui

* blood trail

* can't do blood trails

* how does this even work

* harsher slowdown and more obscured vision

* it really puts it into perspective; you're fucking dying.

* stat_attack

* stop fuckin whispering into radios or whatever it is you're doing

* more fixes

* fix

* fix

* fix the radio shit

* bikeshed?

tgstation-server committed:
Automatic changelog generation for PR #30262 [ci skip]

vuonojenmustaturska committed:
removed superfluous comment

vuonojenmustaturska committed:
removed superfluous comment

vuonojenmustaturska committed:
Rebase

vuonojenmustaturska committed:
Merge branch 'tgui' of https://github.com/vuonojenmustaturska/tgstation into tgui

# Conflicts:
#	tgui/assets/tgui.js

vuonojenmustaturska commented:
jesus christ

vuonojenmustaturska commented:
oh god

ShizCalev commented:
memes
